### PR TITLE
Automate clusterautoscaler test case

### PR DESCRIPTION
**Pull request by jhou1**
Automated clusterautoscaler test case, @sunzhaohua2 @miyadav please review.

Tested locally.
```
      [09:39:09] INFO> === End After Scenario: Cluster should automatically scale up and scale down ===

1 scenario (1 passed)
19 steps (19 passed)
15m47.689s
```

### CODE REVIEW

**Comment by jhou1** [comment]
@akostadinov @pruan-rht Can you please review?

File: features/machine/autoscaler.feature [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    And admin ensures "machineset-clone" machineset is deleted after scenario
+
+    # Create clusterautoscaler
+    Given I use the "openshift-machine-api" project
+    When I run the :create admin command with:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/cloud/cluster-autoscaler.yml |
+    Then the step should succeed
+    And 1 pods become ready with labels:
+      | cluster-autoscaler=default,k8s-app=cluster-autoscaler |
+    And I register clean-up steps:
```
Comment (liangxia) [inline features/machine/autoscaler.feature]: Perhaps register the clean-up earlier. E.g, if the above pod failed to become ready, then the clean-up is not registered successfully.

File: lib/openshift/machine_set.rb [hunk 2]
```diff
@@ -18,7 +18,7 @@ def available_replicas(user: nil, cached: true, quiet: false)
     def ready?(user: nil, cached: false, quiet: false)
       result = {}
       result[:success] = (desired_replicas(user: user, cached: cached, quiet: quiet) ==
-        available_replicas(user: user, cached: cached, quiet: quiet))
+        available_replicas(user: user, cached: true, quiet: quiet))
```
Comment (liangxia) [inline lib/openshift/machine_set.rb]: Update in step implemention ? I though it should be on step definition, that's line 18

Comment (jhou1) [reply lib/openshift/machine_set.rb]: This is intended. Both methods call `raw_resource` method, the second call could use the cached resource to save time. This is @akostadinov's suggestion when I first implemented it.

File: features/machine/autoscaler.feature (hunk 1 above)
Comment (jhou1) [reply features/machine/autoscaler.feature]: Thanks, addressed that.

File: lib/openshift/machine_set.rb (hunk 2 above)
Comment (liangxia) [reply lib/openshift/machine_set.rb]: Ack

File: features/machine/autoscaler.feature [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    # Create clusterautoscaler
+    Given I use the "openshift-machine-api" project
+    When I run the :create admin command with:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/cloud/cluster-autoscaler.yml |
+    Then the step should succeed
+    And I register clean-up steps:
+    """
+    When I run the :delete admin command with:
+      | object_type       | clusterautoscaler |
```
Comment (pruan-rht) [inline features/machine/autoscaler.feature]: @jhou1 please use the step `I ensure...` instead.

File: features/machine/autoscaler.feature [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    # Create machineautoscaler
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/cloud/machine-autoscaler.yml" replacing paths:
+      | ["metadata"]["name"]               | maotest                 |
+      | ["spec"]["minReplicas"]            | 1                       |
+      | ["spec"]["maxReplicas"]            | 3                       |
+      | ["spec"]["scaleTargetRef"]["name"] | <%= machine_set.name %> |
+    Then the step should succeed
+    And I register clean-up steps:
+    """
+    When I run the :delete admin command with:
```
Comment (pruan-rht) [inline features/machine/autoscaler.feature]: @jhou1 please use the step I ensure... instead.

File: features/machine/autoscaler.feature (hunk 3 above)
Comment (jhou1) [reply features/machine/autoscaler.feature]: Oh, I can remember why I didn't use `I ensure ...` in the first place. the `clusterautoscaler` and `machineautoscaler` are custom resources, we don't have a library defined around it. I'll try implement that.

Comment (pruan-rht) [reply features/machine/autoscaler.feature]: @jhou1 [machine_autoscaler.rb](https://github.com/openshift/verification-tests/blob/master/lib/openshift/machine_autoscaler.rb) is there.  You can use that as a template for `clusterautoscaler`

Comment (jhou1) [reply features/machine/autoscaler.feature]: It's updated now.

**Pull request merged**